packages/components/src/button/style.scss

```diff
@@ -53,24 +53,25 @@
 
 		&:hover:not(:disabled) {
 			background: $components-color-accent-darker-10;
-			color: $white;
+			color: $components-color-accent-inverted;
 		}
 
 		&:active:not(:disabled) {
 			background: $components-color-accent-darker-20;
 			border-color: $components-color-accent-darker-20;
-			color: $white;
+			color: $components-color-accent-inverted;
 		}
 
 		&:focus:not(:disabled) {
-			box-shadow: inset 0 0 0 1px $white, 0 0 0 var(--wp-admin-border-width-focus) $components-color-accent;
+			box-shadow: inset 0 0 0 1px $components-color-background, 0 0 0 var(--wp-admin-border-width-focus) $components-color-accent;
 		}
 
 		&:disabled,
 		&:disabled:active:enabled,
 		&[aria-disabled="true"],
 		&[aria-disabled="true"]:enabled, // This catches a situation where a Button is aria-disabled, but not disabled.
 		&[aria-disabled="true"]:active:enabled {
+			// TODO: This needs to be accent-inverted
```

Comment:
The three TODO comments in this stylesheet flag the parts that rely on Sass functions, which will not work when the underlying values are CSS variables.

There are two ways we can address these:

1. Work with designers and see if any of the Sass function parts can be replaced with the standard gray-100 etc. colors. This is the easiest and preferred way when designing themeable components.
2. In the Button component, access the raw theme color values (not CSS variables) via Context, and calculate the necessary colors in JS.

Reply:
Can you expand a bit on what the ask is here? Is it changing out of the main color?

I like to think that we can reach a point where we have a set of system colors with their known contrast properties in light and dark mode (the grays), and then provide a single accent color to a component, and whatever mode it's in defines what colors get applied. That is, ideally I should never have to feed more than a single color along with the dark or light mode prop to a component, and it handles the rest itself. Is that useful?

Reply:
> Can you expand a bit on what the ask is here? Is it changing out of the main color?

To be clear, I'm not asking for anything specific yet in this PR 🙂 We'll make follow-up PRs with more detail, suggest some alternatives, and ping the design team there for discussion.

As an overview of the problem though, we can no longer easily use Sass functions like this, which require the color to be predetermined:

https://github.com/WordPress/gutenberg/blob/2b7486679db9430d457b679ec128693b756ea9e3/packages/components/src/button/style.scss#L75

☝️ This button's text color needs to be either white or black depending on the arbitrary accent color. But we can't use CSS variables as inputs to Sass functions. Things like `rgba(var(--my-color), 0.4)` are not possible.

https://github.com/WordPress/gutenberg/blob/2b7486679db9430d457b679ec128693b756ea9e3/packages/components/src/button/style.scss#L131-L132

https://github.com/WordPress/gutenberg/blob/2b7486679db9430d457b679ec128693b756ea9e3/packages/components/src/button/style.scss#L246-L252

☝️ Similarly in these two examples, we can't do things like `lighten(var(--gray-700), 5%)`.

> the dark or light mode prop

I'm using the term "dark mode" for convenience, but we are currently working toward a more generic solution which allows consumers to pick their own background color. In theory this means _any color_, but in practice it means consumers should be able to choose their own shade of "dark" when doing "dark mode". An app may want `#0a0a0a`, another may want `#1d1f21`, etc.